### DISCUSSION TITLE
CVRP with time windows for multimodal fleet.

### DISCUSSION THREAD

**Discussion by Luab** [google/or-tools#2652] [Routing (and legacy CP) questions] [answered]
I am working with Python version of ORTools. 
For our use case there are several different vehicle types (automobile, scooter, feet courier). So the resulting time matrix is different for each vehicle type. 
Currently, we are considering splitting the problem into three parts (first route feet couriers, than scooters and than route leftovers using auto). Are there any better solution? Is it possible to create heterogenouse time-matrix, which is dependent on vehicle type? Also, is it possible to set hard constraints for each vehicle type independently (for example, do not assign orders, which are too far away to feet couriers)? Thanks!

**Comment by Mizux** [comment, marked as answer]
https://gist.github.com/Mizux/6d9283388150020df7542adf63577786
